### PR TITLE
SetProperties step

### PR DESCRIPTION
**Pull request by tardyp**
## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the master/buildbot/newsfragment directory (and read the README.txt in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

File: master/docs/manual/cfg-properties.rst [hunk 1]
```diff
@@ -431,3 +437,43 @@ For this you can use a special custom renderer as following::
...
+:class:`SetProperties` step
++++++++++++++++++++++++++++
+
+Sometimes you need to compute some properties based on other properties to reuse them in several other steps.
+the
+For this you can use a special custom renderer as following:
+
+.. code-block:: python
+
+    """Example borrowed from Julia's master.cfg"""
```
Comment (tomprince) [inline master/docs/manual/cfg-properties.rst]: What is the license of the config?

Comment (tardyp) [inline master/docs/manual/cfg-properties.rst]: MIT: https://github.com/staticfloat/julia-buildbot

@staticfloat can confirm

**Comment by tardyp** [comment]
> This doesn't appear to be any different from buildbot.steps.master.SetProperty, other than handling multiple properties.

Indeed. I wasn't aware of it. I think it still makes sense given @staticfloat use case, but I need to refactor it in master.py

File: master/docs/manual/cfg-properties.rst (hunk 1 above)
Comment (staticfloat) [inline master/docs/manual/cfg-properties.rst]: Confirmed

**Comment by tardyp** [comment]
@tomprince I did the refactor to make SetProperty and SetProperties more similar.

looks good?

**Pull request merged**